cmd/ingester/app/flags.go

```diff
@@ -42,6 +42,14 @@ const (
 	SuffixGroupID = ".group-id"
 	// SuffixEncoding is a suffix for the encoding flag
 	SuffixEncoding = ".encoding"
+	// SuffixTLS is a suffix for the tls flag
+	SuffixTLS = ".tls"
+	// SuffixCert is a suffix for the tls certificate path flag
+	SuffixCert = ".tls.cert"
+	// SuffixKey is a suffix for the tls key path flag
+	SuffixKey = ".tls.key"
+	// SuffixCA is a suffix for the tls ca path flag
+	SuffixCA = ".tls.ca"
```

Comment:
You're right; i will do that

Reply:
There seems to be no `ZipkinThriftMarshaller`. That means the encoding has different options for consumer and producer applications.  Would something like that be okay for you: 
```
// Initialize implements storage.Factory
func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
	f.metricsFactory, f.logger = metricsFactory, logger
	logger.Info("Kafka factory",
		zap.Any("producer builder", f.Builder),
		zap.Any("topic", f.options.config.Topic))
	p, err := f.NewProducer()
	if err != nil {
		return err
	}
	f.producer = p
	switch f.options.config.Encoding {
	case config.EncodingProto:
		f.marshaller = newProtobufMarshaller()
	case config.EncodingJSON:
		f.marshaller = newJSONMarshaller()
	case config.EncodingZipkinThrift:
		return fmt.Errorf("producers do not support '%s' yet", config.EncodingZipkinThrift)
	default:
		return errors.New("kafka encoding is not one of '" + config.EncodingJSON + "' or '" + config.EncodingProto + "'")
	}
	return nil
}
```

Reply:
I'm fine with that. When calling AddFlags on the shared config, you may want to pass a list of allowed encodings so that the help text doesn't say zipkin for producers